pkg/monitortests/clusterversionoperator/clusterversionchecker/monitortest.go

```diff
@@ -0,0 +1,280 @@
+package clusterversionchecker
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	clientconfigv1 "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortestlibrary/platformidentification"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+type snapshot struct {
+	when time.Time
+	cv   *configv1.ClusterVersion
+	err  error
+}
+
+type monitor struct {
+	collectionDone chan struct{}
+
+	clusterversion []snapshot
+
+	notSupportedReason error
+
+	configv1client *clientconfigv1.Clientset
+}
+
+func NewClusterVersionChecker() monitortestframework.MonitorTest {
+	return &monitor{
+		collectionDone: make(chan struct{}),
+		clusterversion: make([]snapshot, 0, 60), // expect at least 60 snaphots in a job, one per minute
+	}
+}
+
+func (w *monitor) PrepareCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	kubeClient, err := kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
+	if err != nil {
+		return fmt.Errorf("unable to determine if cluster is MicroShift: %v", err)
+	}
+	if isMicroShift {
+		w.notSupportedReason = &monitortestframework.NotSupportedError{Reason: "platform MicroShift not supported"}
+		return w.notSupportedReason
+	}
+	configClient, err := clientconfigv1.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	w.configv1client = configClient
+
+	isSelfManagedHA, err := exutil.IsSelfManagedHA(ctx, configClient)
+	if err != nil {
+		return fmt.Errorf("unable to determine if cluster is HA: %v", err)
+	}
+
+	if !isSelfManagedHA {
+		w.notSupportedReason = &monitortestframework.NotSupportedError{Reason: "non-HA cluster not supported"}
```

Comment:
We can cover HyperShift and single-node with this same tooling too, right?  Are you just excluding them for now, to step into this gently?  Or do you think they should be excluded long-term for reasons I'm not yet clear on?

Reply:
Removed the skipping.

I was trying to cover the new rule in https://github.com/openshift/api/blob/bc748ef9703ff4bfca4c377fb3dd5dc53f955b2b/config/v1/types_cluster_operator.go#L168-L170
Oh. HA-clusters was removed there as well.

It should be working as long as the test can access CV.
We indeed have retries on getting CV, so it should be fine with Single-Node too.